src/store/todoSlice.ts

```diff
@@ -0,0 +1,48 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+export interface Todo {
+  id?: number;
+  task: string;
+  isCompleted: boolean;
+}
+
+interface TodoState {
+  todos: Todo[];
+  todoItem: Todo | undefined;
+}
+
+const initialState: TodoState = {
+  todos: [],
+  todoItem: undefined,
+};
+
+export const todoSlice = createSlice({
+  name: "todoList",
+  initialState,
+  reducers: {
+    addTodo: (state, action: PayloadAction<Todo>) => {
+      state.todos.push(action.payload);
+    },
+    deleteTodo: (state, action: PayloadAction<number>) => {
+      state.todos = state.todos.filter((todo) => todo.id !== action.payload);
+    },
+    setTodoList: (state, action: PayloadAction<Todo[]>) => {
+      state.todos = action.payload;
+    },
+    setTodoItem: (state, action: PayloadAction<Todo | undefined>) => {
+      state.todoItem = action.payload;
+    },
```

Comment:
these look like normal set-state functions. In case of setTodoList you should use getTodos which gets todos from the db. And in case of setTodoItem you should use getTodo whcih gets todo by id.